### DISCUSSION TITLE
Composition API breaks Front-End Packages

### DISCUSSION THREAD

**Discussion by muhammadsaeedparacha** [laravel/nova-issues#6076] [Q&A]
Laravel Nova has tons of packages that provide front-end functionality by overriding it's vue components.
They are breaking as a result of moving to composition API.

Example:
1) Create a Tool
`php artisan nova:tool HeaderPackage`

2) Create a Vue Component to override existing - Main Header in this case, and extend
```
<template>
   // Custom HTML, copy paste existing for testing. purposes
</template>
<script>
import { Button } from 'laravel-nova-ui';
import MainHeader from '@/layouts/MainHeader';

export default {
    components: { CompanyMenu, Button },
    extends: MainHeader,
    setup(props, ctx) {
        return {
            ...MainHeader.setup(props, ctx),
        };
    },
};
</script>

```

The component fails to extend properly, and properties and methods from the parent component are not available. They work in dev version of files `npm run dev` but not in prod `npm run prod`.

Vue Documentation:
Composition API doesn't properly support extending components.
https://vuejs.org/api/options-composition.html
<img width="678" alt="Screenshot 2023-11-29 at 11 12 29 PM" src="https://github.com/laravel/nova-issues/assets/2680384/180ce501-482e-43d9-aa82-4ccd3f8aab9f">

**Comment by annerland** [comment]
+ 1, I've been tried to use composition API on nova custom tools and it also doesn't work for me
